src/lib/reasoners/intervals.mli

```diff
@@ -140,6 +140,10 @@ val add_explanation : t -> Explanation.t -> t
 
 val equal : t -> t -> bool
 
+val pick : to_max:bool -> t -> Numbers.Q.t option
+(** [pick ~to_max t] returns an elements of the set of intervals [t]. If
+    [to_max] is [true], we pick the largest element of [t], if it exists.
+    We look for the smallest element if [to_max] is [false]. *)
```

Comment:
In the case `t = ]-oo;+oo[`, we don't pick the largest element of `t`

Reply:
As I said, `if it exists` :) Otherwise, we pick an element in the last interval of the set of intervals. In the case `]-oo, +oo[`, we pick `0`.